src/Compilers/Core/Portable/InternalUtilities/Hash.cs

```diff
@@ -171,6 +171,7 @@ internal static int GetFNVHashCode(byte[] data)
             return hashCode;
         }
 
+#if !CODE_STYLE
```

Comment:
@dotnet/roslyn-compiler This is the only compiler-layer change. `ReadOnlySpan<T>` isn't available in the code style layer.

Reply:
The compiler layer is likely to increase its usage of `ReadOnlySpan` in various APIs. So I don't think this solution will work for long.
Can the code style layer add this dependency?
Probably should discuss with @agocke.

---
In reply to: [236421104](https://github.com/dotnet/roslyn/pull/31276#discussion_r236421104) [](ancestors = 236421104)

Reply:
> Can the code style layer add this dependency?

Not for now. We compile and support running against 15.7+.

Reply:
➡️ As a follow up:

> The compiler layer is likely to increase its usage of ReadOnlySpan in various APIs. So I don't think this solution will work for long.

This is not likely to impact the code style layer. We compile against the 15.7 compiler assemblies so the only place where these changes could impact us is linked code files, which is only **Hash.cs** and some pooled objects.